kairon/shared/chat/user_media.py

```
@@ -605,3 +607,113 @@ def create_media_doc(
+        Download WhatsApp media, upload to S3, save DB, and return S3 URL
+        """
+        download_url = None
+        file_path = None
+        headers = {}
+        provider = config.get("bsp_type", "meta")
+        if provider == '360dialog':
+            endpoint = f'https://waba-v2.360dialog.io/{whatsapp_media_id}'
+            headers = {
+                'D360-API-KEY': config.get('api_key'),
+            }
+            resp = requests.get(endpoint, headers=headers, stream=True)
```

Ruff (0.15.1)

[error] Probable use of `requests` call without timeout

(S113)